nixos/modules/services/web-apps/galene.nix

```diff
@@ -0,0 +1,187 @@
+      };
+
+      insecure = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Whether Galene should listen in http or in https. If left as the default
+          value (false), Galene needs to be fed a private key and a certificate.
+        '';
+      };
+
+      certFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "/path/to/your/cert.pem";
+        description = ''
+          Path to the server's certificate. The file is copied at runtime to
+          Galene's data directory where it needs to reside.
+        '';
+      };
+
+      keyFile = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        example = "/path/to/your/key.pem";
+        description = ''
+          Path to the server's private key. The file is copied at runtime to
+          Galene's data directory where it needs to reside.
+        '';
+      };
+
+      httpAddress = mkOption {
+        type = types.str;
+        default = "";
+        description = "HTTP listen address for galene.";
+      };
+
+      httpPort = mkOption {
+        type = types.int;
+        default = 8443;
+        description = "HTTP listen port.";
+      };
+
+      staticRootPath = mkOption {
+        type = types.str;
+        default = "${cfg.package.static}/static";
+        example = "/var/lib/galene/static";
+        description = "Web server directory.";
+      };
+
+      recordingsRootPath = mkOption {
+        type = types.str;
+        default = defaultRecordingsRootPath;
+        example = "/var/lib/galene/recordings";
+        description = "Recordings directory.";
+      };
+
+      dataRootPath = mkOption {
+        type = types.str;
+        default = defaultDataRootPath;
+        example = "/var/lib/galene/data";
+        description = "Data directory.";
+      };
+
+      groupsRootPath = mkOption {
+        type = types.str;
+        default = defaultGroupsRootPath;
+        example = "/var/lib/galene/groups";
+        description = "Web server directory.";
+      };
+
+      package = mkOption {
+        default = pkgs.galene;
+        defaultText = "pkgs.galene";
+        type = types.package;
+        description = ''
+          Package for running Galene.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.insecure || (cfg.certFile != null && cfg.keyFile != null);
+        message = ''
+          Galene needs both certFile and keyFile defined for encryption, or
+          the insecure flag.
+        '';
+      }
+    ];
+
+    systemd.tmpfiles.rules = [ ]
+      ++ optional (!cfg.insecure && cfg.certFile != null && cfg.keyFile != null)
+      ''
+        C '${cfg.dataRootPath}/key.pem' 0700 ${cfg.user} ${cfg.group} - ${cfg.keyFile}
+        C '${cfg.dataRootPath}/cert.pem' 0700 ${cfg.user} ${cfg.group} - ${cfg.certFile}
+      ''
```

Comment:
For a couple reasons I'll mention below this probably isn't the best way to copy these files. I would suggest doing a copy in `systemd.services.galene.preStart` or `systemd.services.galene.serviceConfig.ExecStart` and prefix the script with a `+` so it has full permissions.

Reasons include:
- `tmpfiles` **only** runs periodically, or after system activation
- if ownershipt of the key and cert are `root` I'm guessing `tmpfiles` will refuse to copy them... but I haven't verified this